app/src/main/java/com/omelan/cofi/MainActivity.kt

```diff
@@ -121,28 +141,32 @@ class MainActivity : MonetCompatActivity() {
         RecipeDetails(
             recipeId = recipeId,
             onRecipeEnd = { recipe ->
-                lifecycleScope.launch {
-                    db.recipeDao().updateRecipe(recipe.copy(lastFinished = Date().time))
-                    val deepLinkIntent = Intent(
-                        Intent.ACTION_VIEW,
-                        "$appDeepLinkUrl/recipe/$recipeId".toUri(),
-                        this@MainActivity,
-                        MainActivity::class.java,
-                    )
-                    val shortcut =
-                        ShortcutInfoCompat.Builder(this@MainActivity, recipeId.toString())
-                            .setShortLabel(recipe.name)
-                            .setLongLabel(recipe.name)
-                            .setIcon(
-                                IconCompat.createWithResource(
-                                    this@MainActivity,
-                                    recipe.recipeIcon.icon,
-                                ),
-                            )
-                            .setIntent(deepLinkIntent)
-                            .build()
+                if (!isAppInstalled("com.omelan.cofi")) {
+                    showInstallPrompt()
+                } else {
```

Comment:
```suggestion
                if (isInstantApp()) {
                    showInstallPrompt()
                }
```
It should still save latest done recipe in addition to prompting for install